Xamarin.Forms.Core/RadioButton.cs

```diff
@@ -0,0 +1,185 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Xamarin.Forms.Platform;
+
+namespace Xamarin.Forms
+{
+	[RenderWith(typeof(_RadioButtonRenderer))]
+	public partial class RadioButton : Button, IElementConfiguration<RadioButton>
+	{
+		readonly Lazy<PlatformConfigurationRegistry<RadioButton>> _platformConfigurationRegistry;
+
+		static Dictionary<string, List<WeakReference<RadioButton>>> _groupNameToElements;
+
+		public const string IsCheckedVisualState = "IsChecked";
+
+		public static readonly BindableProperty IsCheckedProperty = BindableProperty.Create(
+			nameof(IsChecked), typeof(bool), typeof(RadioButton), false, propertyChanged: (b, o, n) => ((RadioButton)b).OnIsCheckedPropertyChanged((bool)n));
+
+		public static readonly BindableProperty GroupNameProperty = BindableProperty.Create(
+			nameof(GroupName), typeof(string), typeof(RadioButton), null, propertyChanged: (b, o, n) => ((RadioButton)b).OnGroupNamePropertyChanged((string)o, (string)n));
+
+		public static readonly BindableProperty ButtonSourceProperty = BindableProperty.Create(
+			nameof(ButtonSource), typeof(ImageSource), typeof(RadioButton), null);
+
+		public event EventHandler<CheckedChangedEventArgs> CheckedChanged;
+
+		public bool IsChecked
```

Comment:
Do we want to allow tri-state?

Reply:
Personally not a fan, I'll throw it out there in todays sync

Reply:
As discussed; tri-state won't be a thing (for now)